teamcity-snyk-security-plugin-server/src/main/java/io/snyk/plugins/teamcity/server/SnykSecurityReportTab.java

```diff
@@ -52,11 +55,13 @@ protected boolean isAvailable(@NotNull HttpServletRequest request, @NotNull SBui
     return buildType.getRunnerTypes().contains(SnykSecurityRunnerConstants.RUNNER_TYPE);
   }
 
-  private String readSnykHtmlReport(SBuild build) {
+  private String readSnykHtmlReport(@NotNull SBuild build) {
     String snykHtmlReportPath = TEAMCITY_ARTIFACTS_DIR + separator + SNYK_ARTIFACTS_DIR + separator + SNYK_REPORT_HTML_FILE;
     BuildArtifact artifact = build.getArtifacts(BuildArtifactsViewMode.VIEW_HIDDEN_ONLY).getArtifact(snykHtmlReportPath);
+
     if (artifact == null) {
-      return null; // TODO - Handle the case when artifact is not found
+      LOG.error("An error occurred while reading the Snyk HTML report - artifact not found");
```

Comment:
This could just throw an Exception that is caught below - so the exception handling is not duplicated